test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/DefaultBindingMetadataProviderTest.cs

```diff
@@ -147,5 +147,350 @@ public void GetBindingDetails_FindsBindingSource_IfNullFallsBack()
+        {
+            // Arrange
+            var propertyAttributes = new object[]
+            {
+                new BindingBehaviorAttribute(BindingBehavior.Required),
+            };
+
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForProperty(typeof(int), "Length", typeof(string)),
+                new ModelAttributes(propertyAttributes, typeAttributes: new object[0]));
+
+            var provider = new DefaultBindingMetadataProvider();
+
+            // Act
+            provider.GetBindingMetadata(context);
+
+            // Assert
+            Assert.True(context.BindingMetadata.IsBindingAllowed);
+            Assert.True(context.BindingMetadata.IsBindingRequired);
+        }
+
+        [Fact]
+        public void GetBindingDetails_FindsBindRequired_OnProperty()
+        {
+            // Arrange
+            var propertyAttributes = new object[]
+            {
+                new BindRequiredAttribute(),
+            };
+
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForProperty(typeof(int), "Length", typeof(string)),
+                new ModelAttributes(propertyAttributes, typeAttributes: new object[0]));
+
+            var provider = new DefaultBindingMetadataProvider();
+
+            // Act
+            provider.GetBindingMetadata(context);
+
+            // Assert
+            Assert.True(context.BindingMetadata.IsBindingAllowed);
+            Assert.True(context.BindingMetadata.IsBindingRequired);
+        }
+
+        // These attributes have conflicting behavior - the 'required' behavior should be used because
+        // of ordering.
+        [Fact]
+        public void GetBindingDetails_UsesFirstAttribute()
+        {
+            // Arrange
+            var propertyAttributes = new object[]
+            {
+                new BindingBehaviorAttribute(BindingBehavior.Required),
+                new BindNeverAttribute(),
+            };
+
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForProperty(typeof(int), "Length", typeof(string)),
+                new ModelAttributes(propertyAttributes, typeAttributes: new object[0]));
+
+            var provider = new DefaultBindingMetadataProvider();
+
+            // Act
+            provider.GetBindingMetadata(context);
+
+            // Assert
+            Assert.True(context.BindingMetadata.IsBindingAllowed);
+            Assert.True(context.BindingMetadata.IsBindingRequired);
+        }
+
+        [Fact]
+        public void GetBindingDetails_FindsBindRequired_OnContainerClass()
+        {
+            // Arrange
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForProperty(typeof(string), "Property", typeof(BindRequiredOnClass)),
+                new ModelAttributes(propertyAttributes: new object[0], typeAttributes: new object[0]));
```

Comment:
I don't really like the idea because it's only used in a 2 very specialized places.  If we end up doing https://github.com/aspnet/Mvc/issues/2426 then I think we should do it.